### ISSUE TITLE
Sidebar not showing tests or debug options in code

### ISSUE THREAD

**Issue by sleepydog90008** [maty21/mocha-sidebar#129]
### Description
Installed the latest Mocha Sidebar, Node and VS Code; Sidebar doesn't show any test and debug/run test doesn't show in the test files. Mocha is throwing errors trying to render the sidebar. Everything is installed with default options.

I see these errors from Log (Window) also.
[2018-08-02 10:05:11.343] [renderer1] [error] Cannot read property 'passed' of undefined: TypeError: Cannot read property 'passed' of undefined
	at _mochaProvider.onDidChangeTreeData (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\extension.js:181:56)
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:97:917)
	at mochaProvider.refreshExplorer (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\mochaProvider.js:99:35)
	at subscriptions.push.vscode.commands.registerCommand (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\extension.js:110:22)
	at e._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:697:706)
	at e.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:698:212)
	at e._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:658:87)
	at e._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:657:779)
	at e._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:656:851)
	at e._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:656:620)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:655:544
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:658:673
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:96:599
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:97:917)
	at u (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:159:721)
	at Socket.<anonymous> (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:159:898)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at readableAddChunk (_stream_readable.js:178:18)
	at Socket.Readable.push (_stream_readable.js:136:10)
	at Pipe.onread (net.js:560:20)
[2018-08-02 10:05:13.867] [renderer1] [error] Cannot read property 'passed' of undefined: TypeError: Cannot read property 'passed' of undefined
	at _mochaProvider.onDidChangeTreeData (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\extension.js:181:56)
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:97:917)
	at mochaProvider.refreshExplorer (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\mochaProvider.js:99:35)
	at Timeout.setTimeout [as _onTimeout] (C:\Users\xxxxx\\.vscode\extensions\maty.vscode-mocha-sidebar-0.20.17\changesNotification.js:93:57)
	at ontimeout (timers.js:386:14)
	at tryOnTimeout (timers.js:250:5)
	at Timer.listOnTimeout (timers.js:214:5)

#### Steps to Reproduce
1. Install VS Code
2. Install Node
3. Install Mocha Sidebar
4. No test shown in sidebar

#### Expected Results
Test show in sidebar/have Debug/Run in test code

#### Actual Results
Test do not show in sidebar, no option to debug/run test from test file

#### Settings file
// code coverage with istanbul 
  "mocha.coverage": {
    "enable": true,
    "decoration": true,
    "runWithInterval": false,
    "autoUpdateInterval": 20000,
    "activeOnStart": true,
    "runAfterTest": false,
    "runCoverageAfterFileSave": false,
    "reporters": []
  },

  // Mocha: Environment variables to run your tests
  "mocha.env": {},

  // Mocha: Glob to search for test files
  "mocha.files.glob": "test/**/*.js",

  // Mocha: Globs to ignore when searching for test files
  "mocha.files.ignore": [
    "**/.git/**/*",
    "**/node_modules/**/*"
  ],

  // Mocha: mocha package path then the sideBar installed one for example ../node_modules/mocha
  "mocha.logVerbose": true,

  // Mocha: Options to pass to node executable
  "mocha.node_options": [],

  // Mocha: Options to run Mocha
  "mocha.options": {},

  // Mocha: set other mocha path then the built-in one for example ../node_modules/mocha (only path to the package)
  "mocha.path": "",

  // Array of module required to run Mocha. (for example babel-register, coffee-script/register or ts-node/register)
  "mocha.requires": [],

  // Mocha: run tests on each save
  "mocha.runTestsOnSave": "false",

  // Show popup when having any error while executing test
  "mocha.showErrorPopup": "true",

  // Show in explorer view or in test view(new feature since version 1.23)
  "mocha.showInExplorer": false,

  // Mocha: this option allows you to enable/disable  lens nad decorations and set update threshold 
  "mocha.sideBarOptions": {
    "lens": true,
    "decoration": true,
    "autoUpdateTime": 2000,
    "showDebugTestStatus": true
  },

  // Sets a subdirectory as the root for mocha-sidebar
  "mocha.subdirectory": ""

### Versions

- VScode: v1.25.1
- Mocha SideBar: v0.20.17
- Node Version: v10.7.0

### OS version

- [ ] Linux 
- [ ] Mac
- [X] Windows 10 Pro build 17134.191

### Additional Info
I've tried uninstall and reinstall of VS Code and Mocha Sidebar, multiple times, no go.

I am only able to run tests via launch configuration (launch.json) and using debug window. I previously did not need this.
ex config:
{
            "type": "node",
            "request": "launch",
            "name": "Mocha Tests",
            "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
            "args": [
                "-u",
                "tdd",
                "--timeout",
                "999999",
                "--colors",
                "${workspaceFolder}/test"
            ],
            "internalConsoleOptions": "openOnSessionStart"
        }

Code Coverage is working from the sidebar.

Package.json includes:
"scripts": {
    "test": "istanbul test node_modules/mocha/bin/_mocha"
  },

**Comment by sleepydog90008** [comment]
I went back and installed Node 8.11 and mocha sidebar is working. I guess it's something to do with v10 of Node.

**Comment by maty21** [comment]
Yep it's maybe it related although I can't think of any reason for causing it can you try run the test from mocha side bar workspace folder with node 10

**Comment by sleepydog90008** [comment]
sure, I'll close this for now. Maybe when my company needs to upgrade to v10+ I'll revisit this issue. It could of been something local to my machine, but I don't recalling changing anything that would affect this.

**Comment by ottomata** [comment]
BTW, I seem to be having the same issue on node v10+.

**Comment by maty21** [comment]
Hi I'll try my best to figure out what is the problem . I have a lot of pressure right now at the work but I'll find time soon for debugging it sry for the delay

**Comment by share0326** [comment]
any updates?

**Comment by ABialkevich** [comment]
Hame same problem.

**Comment by it-arjan** [comment]
I only got to see tests after pointing mocha.files.glob to my test files in the workspace settings, (settings.json). In usersettings mocha.files.glob seem to have no effect. 

Node 10 LTS, I never saw errors. In fact no test output in output nor debug window. I created compile error in my test. I did not see it in vscode, but the test stayed red in side bar, same as if it failed. 

Is it possible to see similar test output in vscode as when running mocha from cmd?
